src/App.tsx

```diff
@@ -1,61 +1,179 @@
 import './App.scss';
+import {
+  FormControl,
+  Grid,
+  InputLabel,
+  MenuItem,
+  Paper,
+  Select,
+  SelectChangeEvent,
+  TextField,
+} from '@mui/material';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { useState } from 'react';
+import Button from '@mui/material/Button';
+import { TodoList } from './components/TodoList/TodoList';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './types/User';
+import { Task } from './types/Task';
+
+const tasksFromServer = todosFromServer.map(todo => {
+  return {
+    ...todo,
+    user: { ...usersFromServer.find(user => user.id === todo.userId) },
+  };
+}) as Task[];
 
 export const App = () => {
+  const storageItemsCount: number
+    = JSON.parse(localStorage.getItem('tasks') || '{}').length;
+  const storageSavedTasks: Task[]
+    = JSON.parse(localStorage.getItem('tasks') || '{}');
+
+  const [tasks, setTasks] = useState(storageItemsCount > 3
+    ? storageSavedTasks
+    : tasksFromServer);
+  const [titleTask, setTitleTask] = useState('');
+  const [taskOwnerName, setTaskOwnerName] = useState('0');
+  const [taskOwner, setTaskOwner] = useState<User | {}>({});
+  const [isTitleEmpty, setIsTitleEmpty] = useState(false);
+  const [isTaskOwnerEmpty, setIsTaskOwnerEmpty] = useState(false);
+  const [isTasksSaved, setIsTasksSaved] = useState(true);
+
+  function selectTaskOwner(event: SelectChangeEvent<string>) {
+    setIsTaskOwnerEmpty(false);
+    setTaskOwnerName(event.target.value);
+    setTaskOwner({
+      ...usersFromServer
+        .find(user => user.name === event.target.value),
+    });
+  }
+
+  function getNextId(addedTasks: Task[]): number {
+    addedTasks.sort((taskOne, taskTwo) => taskTwo.id - taskOne.id);
```

Comment:
Avoid mutating array, let's use `Math.max` for this purpose

Reply:
Done!